### PR TITLE
[DOCS] adding JSDocs to `arena.ts`

### PR DESCRIPTION
**Pull request by DustinLin**
## What are the changes?

nothing

## Why am I doing these changes?

was looking through these functions and noticed there were no docs

## What did change?

some docs/comments were added

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?

### CODE REVIEW

File: src/field/pokemon.ts [hunk 1]
```diff
@@ -4222,7 +4222,7 @@ export interface AttackMoveResult {
 }
 
 export class PokemonSummonData {
-  public battleStats: integer[] = [ 0, 0, 0, 0, 0, 0, 0 ];
+  public battleStats: integer[] = [ 0, 0, 0, 0, 0, 0, 0 ]; // [Atk, Def, SpAtk, SpDef, Spd, Acc]
```
Comment (DayKev) [inline src/field/pokemon.ts:4225]:
```suggestion
  public battleStats: integer[] = [ 0, 0, 0, 0, 0, 0, 0 ]; // [Atk, Def, SpAtk, SpDef, Spd, Acc, Eva]
```
Not 100% sure on it being accuracy and then evasion but regardless, it's missing.

Comment (DustinLin) [reply src/field/pokemon.ts:4225]: looking at `data/battle-stat.ts` seems to support your guess

**Review by frutescens — DISMISSED** [review]

File: src/field/arena.ts [hunk 2]
```diff
@@ -301,8 +301,8 @@ export class Arena {
 
   /**
    * Attempts to set a new weather to the battle
-   * @param weather new weather to set of type WeatherType
-   * @param hasPokemonSource is the new weather from a pokemon
+   * @param weather {@linkcode WeatherType} new weather to set of type WeatherType
+   * @param hasPokemonSource boolean if the new weather is from a pokemon
```
Comment (flx-sta) [inline src/field/arena.ts:305]:
```suggestion
   * @param hasPokemonSource true if the new weather is from a pokemon
```

you don’t need to put types into JS doc as we are using ts. This is just redundant

File: src/field/arena.ts [hunk 3]
```diff
@@ -301,8 +301,8 @@ export class Arena {
 
   /**
    * Attempts to set a new weather to the battle
-   * @param weather new weather to set of type WeatherType
-   * @param hasPokemonSource is the new weather from a pokemon
+   * @param weather {@linkcode WeatherType} new weather to set of type WeatherType
```
Comment (flx-sta) [inline src/field/arena.ts:304]:
```suggestion
   * @param weather {@linkcode WeatherType} new {@linkcode WeatherType} to set
```
you don’t need to put types into JS doc as we are using ts. This is just redundant

File: src/field/arena.ts [hunk 4]
```diff
@@ -289,8 +289,8 @@ export class Arena {
 
   /**
    * Sets weather to the override specified in overrides.ts
-   * @param weather new weather to set of type WeatherType
-   * @returns true to force trySetWeather to return true
+   * @param weather {@linkcode WeatherType} new weather to set of type WeatherType
+   * @returns boolean - true to force trySetWeather to return true
```
Comment (flx-sta) [inline src/field/arena.ts:293]:
```suggestion
   * @returns true to force trySetWeather to return true
```
you don’t need to put types into JS doc as we are using ts. This is just redundant

File: src/field/arena.ts [hunk 5]
```diff
@@ -289,8 +289,8 @@ export class Arena {
 
   /**
    * Sets weather to the override specified in overrides.ts
-   * @param weather new weather to set of type WeatherType
-   * @returns true to force trySetWeather to return true
+   * @param weather {@linkcode WeatherType} new weather to set of type WeatherType
```
Comment (flx-sta) [inline src/field/arena.ts:292]:
```suggestion
   * @param weather new {@linkcode WeatherType} to set
```
you don’t need to put types into JS doc as we are using ts. This is just redundant

**Review by flx-sta — CHANGES_REQUESTED** [review]
Some minor NITs

File: src/field/arena.ts (hunk 2 above)
Comment (DustinLin) [reply src/field/arena.ts:305]: I find them quite useful though for referring to types in vscode, for example in this case vscode gives you a direct link to the `WeatherType` enum, without the linkcode you have to take an extra step to search/find it

screenshot

Comment (flx-sta) [reply src/field/arena.ts:305]: No offense but there is no extra steps needed. Look above 
`(method) Arena.trySetWeather(weather:  WeatherType, hasPokemonSource: boolean)`

I understand that you mean the internal link but I didn't remove that for the WeatherType. I'm aware of that feature. I just rephrased it to be less redundant

Comment (DustinLin) [reply src/field/arena.ts:305]: thanks, for clarifying i get you (remove extra text, no the linkcode itself)

**Review by flx-sta — DISMISSED** [review]

**Review by DayKev — CHANGES_REQUESTED** [review]
https://github.com/DustinLin/pokerogue/pull/1

**Comment by DustinLin** [comment]
thanks @DayKev , merged in

**Review by flx-sta — APPROVED** [review]

**Review by frutescens — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Pull request merged**